backend/src/main/java/com/empire/World.java

```diff
@@ -1660,19 +1682,44 @@ void gainIncome() {
 		}
 
 		void churchOpinionChangesDueToStateReligion() {
+			Map<String, Ideology> stateIdeologies = kingdoms.keySet().stream().collect(Collectors.toMap(k -> k, k -> NationData.getStateReligion(k, World.this)));
 			for (String k : kingdoms.keySet()) {
-				if (NationData.getStateReligion(k, World.this).religion != Religion.IRUHAN) getNation(k).goodwill -= 5;
-				if (getNation(k).loyalToCult) getNation(k).goodwill -= 10;
+				Religion religion = stateIdeologies.get(k).religion;
+				if (church.hasDoctrine(Church.Doctrine.ANTIECUMENISM) && religion != Religion.IRUHAN && religion != Religion.NONE) getNation(k).goodwill += Constants.antiecumenismStateOpinion;
+				if (church.hasDoctrine(Church.Doctrine.ANTIAPOSTASY) && getNation(k).loyalToCult) getNation(k).goodwill += Constants.antiapostasyOpinion;
+			}
+			Set<String> iruhanNations = stateIdeologies.entrySet().stream().filter(e -> e.getValue().religion == Religion.IRUHAN).map(e -> e.getKey()).collect(Collectors.toSet());
+			if (church.hasDoctrine(Church.Doctrine.INQUISITION)) {
+				Set<String> vesselNations = stateIdeologies.entrySet().stream().filter(e -> e.getValue() == Ideology.VESSEL_OF_FAITH).map(e -> e.getKey()).collect(Collectors.toSet());
+				for (String k : iruhanNations) {
```

Comment:
If you wanted to go full-on stream mode and replace a bunch of these one-line for-loops you could, as an example say (this looks better when its tabbed nicely):
`iruhanNations.stream().filter(k -> vesselNations.stream().anyMatch(v -> !NationData.isAttackingOnSight(k, v, World.this))).forEach(k -> getNation(k).goodwill += Constants.inquisitionOpinion);`
Admittedly I like to keep the stream nesting to a minimum as a general style (don't cross the streams!) but here I would maybe make an exception.

Reply:
Rewrote it in full streaming style, using Google style indentations (two indentation levels on a line continuation, box-model code layout). Take a look and let me know what you think.

Overall, I still feel somewhat undecided on streams versus imperative control flow. The functional stuff works perfectly for obvious set transformations, etc, but once you start pulling in anyMatch / allMatch and operating on nested streams I think it starts to look a little pretty lisp-y. Whether that's a good or bad thing probably depends on who you ask...

Reply:
Seeing it this way...it looks not good I think, mainly due to the amount of lines they now take up.  I would usually only allow myself to nest streams when the nested one was a quick one-liner.  With this style of formatting I don't think there can be such thing as a one-liner stream.  Hate to say it but if this is the alternative I vote to put back the loops.

Reply:
One alternative that deals with the indentation but not the line count is to eliminate the control flow (the loops) but keep using variables (e.g. vesselNations, or even could define the filter predicate itself as a variable). Still, the operation is complex.

Reply:
Yes, in my original example I was keeping the variables you had already defined.  Personally, I love using streams, however I've shot myself in the foot enough times by overdoing it with them.  The medium approach using variables would be acceptable to me, doing each of the ops in a single statement I think is too much.

Reply:
Named the predicates - I think that's a bit clearer, if more verbose.